java/Oleksii-Riabov/src/com/zoolatech/lecture4/tasks/_5/Task5.java

```diff
@@ -0,0 +1,58 @@
+package com.zoolatech.lecture4.tasks._5;
+
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.util.Scanner;
+
+/**
+ * Write a program that accepts two numbers and a symbol of an operation
+ * (as a character) and outputs the result of the selected operation.
+ * Possible characters for the operation: ‘+’, ‘-’, ‘*’, ‘/’, ‘%’. The
+ * program should work with decimal numbers of any length and precision.
+ */
+
+public class Task5 {
+    public static void main(String[] args) {
+        Scanner scanner = new Scanner(System.in);
+
+        BigDecimal number1;
+        BigDecimal number2;
+
+        char operator;
+
+        if(scanner.hasNextBigDecimal()){
+            number1 = scanner.nextBigDecimal();
+        } else {
+            throw new IllegalArgumentException("Enter numeric value");
+        }
+
+        if(scanner.hasNextBigDecimal()){
+            number2 = scanner.nextBigDecimal();
+        } else {
+            throw new IllegalArgumentException("Enter numeric value");
+        }
+
+        if(scanner.hasNext()){
+            operator = scanner.next().charAt(0);
+        } else {
+            throw new IllegalArgumentException("Enter character value");
+        }
+
+        numbersEvaluationCaseMethod(number1, number2, operator);
+    }
+
+    public static void numbersEvaluationCaseMethod(BigDecimal number1, BigDecimal number2, char c) {
+        switch (c) {
+            case '+' -> printResult(number1.add(number2));
```

Comment:
you could use the switch expression and move a call to `printResult` out of the switch block

Reply:
Done.